src/main/java/pl/lukakan/contactform/MainController.java

```diff
@@ -0,0 +1,67 @@
+package pl.lukakan.contactform;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+import org.thymeleaf.TemplateEngine;
+import org.thymeleaf.context.Context;
+
+import javax.mail.MessagingException;
+
+@Controller
+public class MainController {
+
+    private final MailService mailService;
+    private final TemplateEngine templateEngine;
+    private final MessageConfiguration messageConfiguration;
+
+    @Autowired
+    public MainController(MailService mailService, TemplateEngine templateEngine, MessageConfiguration messageConfiguration) {
+        this.mailService = mailService;
+        this.templateEngine = templateEngine;
+        this.messageConfiguration = messageConfiguration;
+    }
+
+    @GetMapping("/")
+    public String homePage() {
+        return "home";
+    }
+
+    @GetMapping("/contact")
+    public String contactPage(Model model) {
+        model.addAttribute("message", new Message());
+        return "contact";
+    }
+
+    @PostMapping("/send")
+    public String sendMessage(Message message, RedirectAttributes redirectAttributes) {
+        Message ownerMessage = prepareMessageBody(message, messageConfiguration.getOwnerMailHeader(), message.getContent());
```

Comment:
Wydaje mi się, że zbudowanie wspólnej obsługi budowania wiadomości dla właściciela i użytkownika zaburza trochę czytelność. Dodatkowo zawartość tych dwóch wiadomości może znacząco odbiegać od siebie w przyszłości, myślę że można pójść w stronę rozbicia tych dwóch zadań do osobnych metod, stworzyć dwa osobne szablony dla obu wiadomości i nawet zawrzeć teksty, ktore trzymasz w propertiesach (*header i *content) bezpośrednio w szablonach. Chyba że jest jakiś inny powód, dla którego chciałbyś mieć te teksty w propertiesach, to daj znać wtedy :P 
Pamiętaj, że wyciąganie wspólnych fragmentów kodu do jednej metody nie zawsze jest dobre ;) Istnieją m.in. 2 przesłanki, które mogą zasugerować by tego nie robic:
* takie działanie sprawia, że kkod jest **dużo** mniej czytelny i **dużo** ciężej prześledzić jego działanie.
* prawdopodobnie w niedalekiej przyszłości działanie funkcjonlaności z wspólną częścią kodu rozbiegnie się tak bardzo, że i tak trzeba będzie pozbyć się wspólnej części

Reply:
Jasne. Generalnie chciałem umieścić header i content w propertisach jako wartośc łatwiej edytowalną + znowu sie zafixowalem na reużywalności kodu -> koniecznie chcialem zrobic jeden template, ale fakt, że w tym wypadku to bez sensu